src/main/java/objects/inventory/InventoryState.java

```diff
@@ -0,0 +1,137 @@
+import interfaces.State;
+import objects.item.*;
+import objects.character.Player;
+
+/**
+ * Use case to change inventory based on user input
+ */
+public class InventoryState implements State {
+    /**
+     * inventory: entity for inventory
+     * isDone: whether the state is done
+     * awaitingInput: whether the state is waiting for user input
+     * addItemCommand: user command when user want to add item
+     * removeItemCommand: user command when user want to remove item
+     * useItemCommand: user command when user want to use item
+     * quitCommand: user command when user want to quit inventory
+     */
+    private Inventory inventory;
+    private Player player;
+    private boolean isDone = false;
+    private boolean awaitingInput = false;
+    private final String addItemCommand;
+    private final String removeItemCommand;
+    private final String useItemCommand;
+    private final String quitCommand;
+
+    /**
+     * @param inventory inventory contains items
+     */
+    public InventoryState(Inventory inventory, Player player){
+
+        this.inventory = inventory;
+        this.player = player;
+        this.quitCommand = "quit";
+        this.removeItemCommand = "remove";
+        this.addItemCommand = "add";
+        this.useItemCommand = "use";
+    }
+
+    /**
+     * Executes when not awaiting input
+     */
+    @Override
+    public void preInput() {
+        awaitingInput = true;
+        // use outputHandler to display inventory, call inventory.viewInventory()
+        // use outputHandler to display choices of commands below
+        // Commands:
+        // Use item: type "use i" ("i" is the i-th item in inventory)
+        // Remove item: type "remove i" ("i" is the i-th item in inventory)
+        // Add item: type "add itemType" (itemType: Sword, Potion, Armor)
+    }
+
+    /**
+     * Executes when awaiting input from the user
+     * @param input from the user
+     */
+    @Override
+    public void postInput(String input) {
+        String[] inputarr = input.split(" ");
+        String command = inputarr[0];
+        if (command.equals(quitCommand)) {
+            isDone = true;
+            return;
+        }
+        boolean success = false;
+        if (command.length() == 2) {
+            if (command.equals(removeItemCommand)) {
+                success = inventory.removeItem(Integer.parseInt(inputarr[1]));
+//            } else if (command.equals(addItemCommand)) {
+//                Item item = inputToItem(player.getLevel(), inputarr[1]);
+//                success = inventory.addItem(item);
+//                if (!success){
+//                    // say inventory is full
+//                }
+            } else if (command.equals(useItemCommand)) {
+                String ability = inventory.useItem(Integer.parseInt(inputarr[1]));
+                if (ability != null){
+                    success = true;
+                    // return the ability to battle system
```

Comment:
I dont think this is possible
you would need to switch managers to do this, and it would be a lot easier if this was its own state in battleManager like useItemState